### ISSUE TITLE
Add build-and-install tooling for developing Kedro for Databricks and documentation on how to use it

### ISSUE THREAD

**Issue by jmholzer** [kedro-org/kedro#2318]
## Description
As part of the Databricks work (including that under https://github.com/kedro-org/kedro/issues/2185), we should add tooling that automates the manual process of building and installing Kedro on Databricks and add a section to our documentation that guides contributors in using it.

## Context
This is an important precursor for other tickets in the Databricks milestone as it will enable developers and reviewers to develop Kedro for Databricks efficiently.

**Comment by stichbury** [comment]
I don't quite understand why this is something under "Contribute to Kedro" documentation. 

* Are we specifically looking for contributors to help us with the databricks content? 
* Or do we specifically need to help people that want to contribute to Kedro to use databricks for their contribution effort?

Sorry if it's obvious -- I think I probably need more understanding of what the docs would look like for this issue to be completed.

**Comment by jmholzer** [comment]
Hi @stichbury, I'm sorry for the delay in getting back to you, I only read your comment now.

While I was working on a [PR](https://github.com/kedro-org/kedro/pull/2369) for this ticket and I feel that I originally defined it incorrectly, I'm sorry about that.

The challenge is to help people who develop Kedro do so for fixes and features on Databricks. I originally intended it to purely be a guide for setting up a development environment, though I ended up adding a fair amount of custom tooling to speed up development also. I have added the new guide under the 'Contribute to Kedro' section of our docs, though it may be better to add it somewhere else.

I will schedule a call with you and the other reviewer on this PR to take you through it (if you would like), as I understand it could be a bit confusing 🙂.
